src/components/SignUpScreen/SignUpScreen.js

```diff
@@ -30,42 +30,46 @@ export default function SignUp() {
     const form = event.currentTarget;
     const isValid = form.checkValidity();
 
-    if (isValid) {
-      const data = new FormData(form);
-      const name = data.get('firstName');
-      const lastName = data.get('lastName');
-      const password = data.get('password');
-      const retypePassword = data.get('confirmPassword');
+    if (!isValid) {
+      console.log('not valid fields'); // todo add logic
+      return;
+    }
 
-      const isPasswordEqual = password === retypePassword;
+    const data = new FormData(form);
+    const name = data.get('firstName');
+    const lastName = data.get('lastName');
+    const password = data.get('password');
+    const retypePassword = data.get('confirmPassword');
 
-      if (isPasswordEqual) {
-        const userData = {
-          email,
-          first_name: name,
-          last_name: lastName,
-          password,
-          retypePassword: password,
-        };
+    const isPasswordEqual = password === retypePassword;
 
-        dispatch(setUser({
-          name,
-          lastName,
-          password
-        }));
-        register(userData)
-          .then((data) => {
-            if (data) {
-              navigate('../login');
-            }
-          })
-          .catch((err) => console.log(err)); // todo add logic
-      } else {
-        console.log('passwords are not the same'); // todo add logic
-      }
-    } else {
-      console.log('not valid fields'); // todo add logic
+    if (!isPasswordEqual) {
```

Comment:
Might be better now (password === retypePassword)

Reply:
password !== retypePassword added